### PR TITLE
[aboutsettings] Expose adaptationVersion (Contributes to JB#13494)

### PR DESCRIPTION
**Pull request by thp**

### CODE REVIEW

File: src/aboutsettings.cpp [hunk 1]
```diff
@@ -39,18 +39,23 @@
...
+    , m_devinfo(new QDeviceInfo(this))
+    , m_ssuinfo(new SsuDeviceInfo())
 {
     qDebug() << "Drives:" << m_sysinfo->allLogicalDrives();
 }
 
 AboutSettings::~AboutSettings()
 {
+    if (m_ssuinfo) {
+        delete m_ssuinfo;
```
Comment (pvuorela) [inline src/aboutsettings.cpp]: delete doesn't need null check.

File: src/aboutsettings.cpp [hunk 2]
```diff
@@ -39,18 +39,23 @@
...
 AboutSettings::AboutSettings(QObject *parent)
-    : QObject(parent),
-      m_sysinfo(new QStorageInfo(this)),
-      m_netinfo(new QNetworkInfo(this)),
-      m_devinfo(new QDeviceInfo(this))
+    : QObject(parent)
+    , m_sysinfo(new QStorageInfo(this))
+    , m_netinfo(new QNetworkInfo(this))
+    , m_devinfo(new QDeviceInfo(this))
+    , m_ssuinfo(new SsuDeviceInfo())
```
Comment (pvuorela) [inline src/aboutsettings.cpp]: If aiming to reduce changed lines with comma on start, this kind of fails :)

**Comment by pvuorela** [comment]
In general LGTM.

File: src/aboutsettings.cpp (hunk 2 above)
Comment (thp) [inline src/aboutsettings.cpp]: Yes, for this one time it does ;) Next member variable addition will be a single-line diff, though :)

File: src/aboutsettings.cpp (hunk 1 above)
Comment (thp) [inline src/aboutsettings.cpp]: Will fix

File: src/aboutsettings.cpp [hunk 3]
```diff
@@ -39,6 +39,81 @@
 #include <QFile>
 #include <QByteArray>
 #include <QRegularExpression>
+#include <QMap>
+#include <QTextStream>
+
+
+static QMap<QString, QString> parseReleaseFile(const QString &filename)
+{
+  QMap<QString, QString> result;
```
Comment (pvuorela) [inline src/aboutsettings.cpp]: Inconsistent two space indentation on this function.

Comment (thp) [inline src/aboutsettings.cpp]: Indentation fixed now

File: src/aboutsettings.cpp [hunk 4]
```diff
@@ -39,6 +39,81 @@
...
+                // Remove the quotes
+                value = value.mid(1, value.size() - 2);
+            }
+
+            // "If double or single quotes or backslashes are to be used within
+            // variable assignments, they should be escaped with backslashes,
+            // following shell style."
+            value = value.replace("\\\"", "\"");
+            value = value.replace("\\'", "'");
+            value = value.replace("\\\\", "\\");
```
Comment (pvuorela) [inline src/aboutsettings.cpp]: According to os-release spec there should be more to escape, so previous version unescaping should have been safer.

File: src/aboutsettings.cpp [hunk 5]
```diff
@@ -39,6 +39,81 @@
...
 #include <QRegularExpression>
+#include <QMap>
+#include <QTextStream>
+
+
+static QMap<QString, QString> parseReleaseFile(const QString &filename)
+{
+    QMap<QString, QString> result;
+
+    // Specification of the format, encoding is similar to /etc/os-release
```
Comment (pvuorela) [inline src/aboutsettings.cpp]: "similar to", wouldn't this be exactly os-release?

File: src/aboutsettings.cpp (hunk 4 above)
Comment (pvuorela) [inline src/aboutsettings.cpp]: Alternatively could add here ` and $

Comment (thp) [inline src/aboutsettings.cpp]: So if I change that to:

```
value = value.replace(QRegularExpression("\\\\(.)"), "\\1");
```

Would that do the trick?

Comment (thp) [inline src/aboutsettings.cpp]: Ideally we'd use something like `shlex.split()` in Python, but SO doesn't have a good answer for that: http://stackoverflow.com/questions/21959706/pythons-shlex-alternative-for-c-c

Comment (pvuorela) [inline src/aboutsettings.cpp]: I think the regexp should work, was there before already.

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**